types/react-hyperscript/index.d.ts

```diff
@@ -0,0 +1,16 @@
+// Type definitions for react-hyperscript 3.0
+// Project: https://github.com/mlmorg/react-hyperscript
+// Definitions by: tock203 <https://github.com/tock203>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.3
+
+import { ComponentClass, StatelessComponent, ReactElement } from 'react';
+
+type ReactHyperScript = <P>(
+    componentOrTag: ComponentClass<P> | StatelessComponent<P> | string,
+    properties?: P,
+    children?: ReadonlyArray<ReactElement<any>> | string
+) => ReactElement<P>;
+
+declare const h: ReactHyperScript;
```

Comment:
Instead of declaring a `const` of function type, just declare a function: `declare function h<P>(...)`

Reply:
Thanks for quick response.
Exporting function using `export =` fails with below error because [ES spec does not allow that](https://stackoverflow.com/questions/39415661/what-does-resolves-to-a-non-module-entity-and-cannot-be-imported-using-this).

> react-hyperscript-tests.ts(2,20): error TS2497: Module '"/home/tock/.local/opt/ghq/github.com/DefinitelyTyped/DefinitelyTyped/types/react-hyperscript/index"' resolves to a non-module entity and cannot be imported using this construct.

But, tsc allows exporting function defined by `const`. (I have no idea why it is so)
That's why I define function using const.

Reply:
You can work around it using `namespace h {}`.

Reply:
😲
I did.